index.js

```diff
@@ -107,37 +112,74 @@ function formatTestResults(results, url) {
 async function run() {
     try {
         // Get inputs
-        const apiToken = core.getInput('api-token');
         const suiteId = core.getInput('suite-id');
-        const payloadInput = core.getInput('payload');
+        const suite = core.getInput('suite');
+        const payload = core.getInput('payload');
+        const stories = core.getInput('stories');
+
+        if (suiteId && suite) {
+            core.setFailed('Please provide either suite-id or suite, not both.');
+            return;
+        }
+        if (!suiteId && !suite) {
+            core.setFailed('Please provide either suite-id or suite.');
+            return;
+        }
+
+        if (suiteId && stories) {
+            core.setFailed('When "suite-id" is provided, "stories" should come from "payload", not "stories".');
+            return;
+        }
+
+        if (suite && payload) {
+            core.setFailed('When "suite" is provided, "stories" should come from "stories", not "payload".');
+            return;
+        }
+
+        const apiToken = core.getInput('api-token');
+        const payloadInput = suiteId ? core.getInput('payload') : core.getInput('stories');
         const waitForResults = core.getInput('wait-for-results') || 'yes';
         const domain = core.getInput('domain') || 'https://api.heal.dev';
         const commentOnPr = core.getInput('comment-on-pr') || 'no';
         const githubToken = core.getInput('github-token');
 
         // Parse and validate payload
-        let payload;
+        let validatedPayload;
         try {
-            payload = payloadInput ? JSON.parse(payloadInput) : {};
+            validatedPayload = payloadInput ? JSON.parse(payloadInput) : {};
+            core.info(`Stories: ${JSON.stringify(validatedPayload)}`);
+            if (Array.isArray(validatedPayload)) {
+                validatedPayload = { stories: validatedPayload };
+            }
```

Comment:
If I understand well, this condition is used to detect whether we're loading stories from `core.getInput('payload')` or `core.getInput('stories')`. If so, this intention is too implicit in the code, which may lead to bugs in the future.

Also, this will not work if the structure/schema of `payload` and `story` diverge.

=> I suggest to avoid assigning one or the other in `payloadInput` (cf line 140, above), and doing something like this instead:

```suggestion
            /** @type { { entrypoint?: string, variables?: Record<string, string> }[] } */
            const validatedStories = payloadInput
              ? JSON.parse(core.getInput('payload')) // TODO: use map to conform items to type, if needed
              : { stories: JSON.parse(core.getInput('stories')) }; // TODO: use map to conform items to type, if needed
```